app/mailers/application_mailer.rb

```diff
@@ -36,4 +37,27 @@ def generate_unsubscribe_token(id, email_type)
   def use_custom_host
     ActionMailer::Base.default_url_options[:host] = Settings::General.app_domain
   end
+
+  def perform_deliveries?
+    return unless Rails.env.production?
+
+    ActionMailer::Base.perform_deliveries = Settings::General.smtp_settings["password"].present? ||
+      ENV["SENDGRID_API_KEY"].present?
+    ActionMailer::Base.perform_deliveries
```

Comment:
this class inherits from `ActionMailer::Base` so this should be enough:

```ruby
  def perform_deliveries?
    self.perform_deliveries = Settings::General.smtp_settings["password"].present? ||
      ENV["SENDGRID_API_KEY"].present?
  end
```

is there a reason why we modify the base class's attribute, instead of the class's ?

Reply:
I didn't think this through enough. `ForemInstane.smtp_enabled?` calls this. I updated the code there too to reflect the change.

I applied your suggestion! Thank you again!